back/app/services/localization_service.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class LocalizationService
+  # @param [Integer] hour, 0-23
+  # @return [String] either 12 or 24-hour localized representation, depending on locale
+  def hour_of_day_string(hour, locale)
+    case locale
+    when 'en', 'en-GB', 'en-US', 'en-CA'
+      hour > 12 ? "#{hour % 12} PM" : "#{hour} AM" # e.g. "1 PM"
+    when 'hr-HR' # Croatian
+      hour > 12 ? "#{hour % 12}:00 sati" : "0#{hour}:00 sata" # e.g. "01:00 sata"
```

Comment:
Croatian makes using YAML language files for formatting the time difficult, so this kind of method is probably needed. 

For example, I'm not sure how we'd implement something like [this](https://stackoverflow.com/questions/8532121/displaying-hour-format-am-pm-or-not-in-rails-based-on-current-locale#:~:text=Add%20a%20comment-,5,-Of%20course%20you) for Croatian.

Reply:
Looks like using pre-made YAML files is the way to go here (not this hacky service method). For example [these](https://github.com/svenfuchs/rails-i18n/tree/master/rails/locale), which are referenced by Rails docs [here](https://guides.rubyonrails.org/v6.1/i18n.html#adding-date-time-formats).